src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java

```diff
@@ -0,0 +1,93 @@
+package com.karankumar.bookproject.backend.service;
+
+import com.karankumar.bookproject.backend.entity.Author;
+import com.karankumar.bookproject.backend.entity.Book;
+import com.karankumar.bookproject.backend.entity.PredefinedShelf;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@SpringBootTest
+public class BookServiceTest {
+
+    private static PredefinedShelfService shelfService;
+
+    private static AuthorService authorService;
+    private static BookService goalService;
+
+    private Book bookWithoutShelf;
+    private Book bookWithoutAuthor;
+    private Book validBook;
+    private Author author;
+
+    @BeforeEach
+    public void setup(@Autowired BookService goalService, @Autowired AuthorService authorService,@Autowired PredefinedShelfService shelfService) {
+
+        List<PredefinedShelf> shelves = shelfService.findAll();
+        PredefinedShelf toRead = shelves.stream()
+                .takeWhile(s -> s.getPredefinedShelfName().equals(PredefinedShelf.ShelfName.TO_READ))
+                .collect(Collectors.toList())
+                .get(0);
+
+
+        this.goalService = goalService;
+        goalService.deleteAll();
+        this.authorService = authorService;
+        authorService.deleteAll();
+
+        author = new Author("Test First Name","Test Last Name");
+        bookWithoutShelf = new Book("Test Title", author);
+        bookWithoutAuthor = new Book("Book without Author",null);
+        validBook = new Book("Book Name",author);
+        validBook.setShelf(toRead);
+    }
+
```

Comment:
Could you delete this line?

Reply:
This line break is between the last method end and the start of other methods. I think it is required. Please confirm.

Reply:
This is an old comment (this was for your first commit -- see https://github.com/knjk04/book-project/blob/8bcc25b14ce13878e4d259c6ffd5a5b2ef0a651a/src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java)